modules/reactor/doc/content/source/meshgenerators/HexagonConcentricCircleAdaptiveBoundaryMeshGenerator.md

```diff
@@ -0,0 +1,40 @@
+# HexagonConcentricCircleAdaptiveBoundaryMeshGenerator
+
+!syntax description /Mesh/HexagonConcentricCircleAdaptiveBoundaryMeshGenerator
+
+## Overview
+
+The `HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` object generates a 2D mesh for concentric circles with a hexagon enclosure with or without external ducts. It is similar to [`PolygonConcentricCircleMeshGenerator`](/PolygonConcentricCircleMeshGenerator.md) and the majority of the parameters and functionalities of this object are described in the documentation of [`PolygonConcentricCircleMeshGenerator`](/PolygonConcentricCircleMeshGenerator.md). The major unique feature of this object is that the `num_sectors_per_side` parameter will be overridden to enable stitching to another hexagonal mesh (provided via `input` to this object). This object therefore creates hexagonal meshes that can be directly stitched with hexagonal meshes generated by [`PolygonConcentricCircleMeshGenerator`](/PolygonConcentricCircleMeshGenerator.md) and [`PatternedHexMeshGenerator`](/PatternedHexMeshGenerator.md). This object only works for hexagons instead of general polygons.
+
+### More Information
+
+In order to adapt the external boundary to match input meshes (typically generated by [`PatternedHexMeshGenerator`](/PatternedHexMeshGenerator.md)), two extra input parameters must be provided:
+
+- `sides_to_adapt`: a vector parameter that can include up to six elements which lists the reference hexagon side numbers to adaptively mesh. These entries must be non-repeating integers from 0 to 5 (see [Figure 1](#schematic)). The hexagon sides are indexed counterclockwise starting from the upper-right side.
+- `inputs`: a vector parameter that includes the same number of elements as `sides_to_adapt`. These elements are input hexagonal meshes whose boundary nodes will be queried and used to adaptively mesh a side of the hexagonal mesh being created. The input hexagonal meshes are generated by [`PatternedHexMeshGenerator`](/PatternedHexMeshGenerator.md) or [`PolygonConcentricCircleMeshGenerator`](/PolygonConcentricCircleMeshGenerator.md). Multiple meshes may be input, each one corresponding to the side index value specified in `sides_to_adapt`. The same input mesh may be repeated multiple times.
+
+!media media/meshgenerators/polygon.png
+      style=display: block;margin-left:auto;margin-right:auto;width:40%;
+      id=schematic
+      caption=A schematic drawing showing the hexagon side indices in `HexagonConcentricCircleAdaptiveBoundaryMeshGenerator` object.
+
+The `hexagon_size` provided here must be consistent with the `hexagon_size` values of all the `inputs` generators.
+
+One major application of this object is to generate meshes for control drums. In that case, `is_control_drum` can be set as true so that corresponding `MeshMetaData` can be generated for future use with control drum related objects.
```

Comment:
```suggestion
One major application of this object is to generate meshes for control drums. In that case, [!param](/Mesh/HexagonConcentricCircleAdaptiveBoundaryMeshGenerator/is_control_drum) can be set as true so that corresponding `MeshMetaData` can be generated for future use with control drum related objects.
```